src/containers/LocationPage/LocationPage.jsx

```diff
@@ -491,14 +431,22 @@ class LocationPage extends PureComponent {
   renderClientIspSelector() {
     const { topClientIsps = [], selectedClientIspInfo } = this.props;
 
+    // Create ASN Number to ISP Object dictionary
```

Comment:
I am wondering if this code already exists somewhere else and can be leveraged here... Why does this dictionary need to be made? I think that @ Pascal/ @ruijieyew may have looked into this before.

Reply:
Yeah this was something I was unsure about. The IspSelectWithIncidents used to take in a list of objects. So instead I thought maybe passing in a dictionary was slightly better because we would only need to iterate through the list once (to create the dictionary). When using just a list of incidents, there were times were we needed to search the entire list to get the specific corresponding object (lines 135 and 148 in IspSelectWithIncidents.jsx). Although, at one point we need to get the list of objects (line 80 in IspSelectWithIncidents.jsx). Maybe it would make more sense just to leave it as a list of objects to be more consistent?

Reply:
Clarified offline, dictionary seems good!